imports/plugins/included/connectors-shopify/server/hooks/orders.js

```diff
@@ -0,0 +1,278 @@
+  const order = Orders.findOne(doc._id); // only this object has the original transforms defined
+  const shops = Object.keys(order.getItemsByShop());
+  shops.forEach((shopId) => {
+    Orders.update({ _id: order._id }, {
+      $push: {
+        exportHistory: {
+          status: "failed",
+          dateAttempted: new Date(),
+          exportMethod: "reaction-connectors-shopify",
+          shopId: shopId
+        }
+      }
+    });
+  });
+}
+
+/**
+ * @private
+ * @summary build a new object for export to Shopify
+ * @param {Object} doc - the order to convert
+ * @param {Number} index - the sequence number of billing/shipping records we are processing
+ * @param {String} shopId - the Id of the shop we are processing for
+ * @returns {Object} shopifyOrder - the converted order
+ */
+function convertRcOrderToShopifyOrder(doc, index, shopId) {
+  const order = Orders.findOne(doc._id); // only this object has the original transforms defined
+  const shopifyOrder = {};
+  const billingAddress = convertAddress(order.billing[index].address);
+  shopifyOrder.billing_address = billingAddress;
+  const shippingAddress = convertAddress(order.shipping[index].address);
+  shopifyOrder.shipping_address = shippingAddress;
+  shopifyOrder.customer = convertCustomer(billingAddress, order);
+  shopifyOrder.email = order.email;
+  const paymentType = order.billing[index].method;
+  if (paymentType === "credit" && order.billing[index].mode === "authorize") {
+    shopifyOrder.financial_status = "authorized";
+  } else {
+    shopifyOrder.financial_status = "paid";
+  }
+  shopifyOrder.id = order._id;
+  const itemsForShop = order.getItemsByShop()[shopId];
+  shopifyOrder.line_items = convertLineItems(itemsForShop, order);
+  shopifyOrder.phone = order.billing[index].address.phone;
+  shopifyOrder.source_name = "reaction_export";
+  shopifyOrder.subtotal_price = order.getSubtotalByShop()[shopId];
+  shopifyOrder.token = order._id;
+  shopifyOrder.total_discounts = order.getDiscountsByShop()[shopId];
+  shopifyOrder.total_line_item_price = order.getItemsByShop()[shopId].reduce((total, item) => {
+    return total + (item.variants.price * item.quantity);
+  }, 0);
+  shopifyOrder.total_price = order.getTotalByShop()[shopId];
+  shopifyOrder.total_tax = order.getTaxesByShop()[shopId];
+  shopifyOrder.total_weight = shopifyOrder.line_items.reduce((sum, item) => {
+    return sum + (item.grams * item.quantity);
+  }, 0);
+  return shopifyOrder;
+}
+
+/**
+ * @private
+ * @summary Normalize all weights to grams
+ * @param {Number} weight - the original weight value
+ * @param {String} shopId - the shop we are converting for
+ * @returns {Number} the normalized value
+ */
+function normalizeWeight(weight, shopId) {
+  // if weight is not grams, convert to grams if is grams just return
+  const shop = Shops.findOne(shopId);
+  const { baseUOM } = shop;
+  // Could've used switch, this was easier to read
+  // should these be a utilities module somewhere?
```

Comment:
Yes! Should definitely add these to a util module somewhere. I've thought the same thing several times.